doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+         * @return the compartment counts for each section for each neuron
+         * @version 1.0
+         */
+        const CompartmentCounts& getCompartmentCounts() const;
+
+        /** Get the number of compartments for the given neuron.
+         *
+         * The neuron index is derived from the order in the GID set provided by
+         * updateMapping().
+         *
+         * @param index neuron index per current GID set
+         * @return number of compartments for the given neuron or 0 if the
+         *         report has just been created or updateMapping is called, and
+         *         not frame has been read yet. Also 0 for neurons not part of
+         *         the report at the source.
+         */
+        size_t getNumCompartments( const size_t index ) const;
+
+        // Same as version 1.0
+        float getStartTime() const;
+
+        /** @return the most up to date end time of the report.
+         *
+         * This value may change upon the completion of readNextFrame, seek,
+         * or writeHeader.
+         *
+        float getEndTime() const;
+
+        // Same as version 1.0
+        float getTimestep() const;
+
+        // Same as version 1.0
+        const std::string& getDataUnit() const;
+
+        // Same as version 1.0
+        const std::string& getTimeUnit() const;
+
+        // Same as version 1.0
+        size_t getFrameSize() const;
+
+        /**
+         * @return The current position in the report. This is the start time
+         *         to use in write operations and the next read operation.
+         */
+        float getCurrentTime() const;
+
+        enum State { OK = 0, ENDED = 1, FAILED = 2 };
+        /** @return The state after the last completed operation. */
+        State getState() const;
+
+        /**
+         * Read the next compartment frame.
+         *
+         * Preconditions:
+         * - r.getState() is OK or ENDED.
+         * - The report was open in read mode.
+         * - There is no previous read or seek operation with a pending future.
+         *
+         * Postconditions:
+         * Let:
+         *  - r be the CompartmentReport
+         *  - f be the returned future by r.readNextFrame()
+         *  - and s = r.getCurrentTime() before read is called (s = -inf if
+         *    undefined):
+         * After f.wait() returns the following are all true:
+         *  - If r.getState() == ENDED s == r.getCurrentTime() and f.get() is
+         *    empty
+         *  - Otherwise r.getCurrentTime() ~= s + r.getTimestep() (the timestamp
+         *    is rounded to the nearest multiple of the timesetp).
+         *  - The mapping data becomes defined.
+         *
+         * After successful f.wait_for or f.wait_until, the result is the same
+         * as above. If they time out, no observable state changes.
+         *
+         * If the state is FAILED or some other operation is still pending
+         * when read is called the result is undefined.
```

Comment:
For consistency with the SpikeReport it should throw logic_error if preconditions are not met instead of having an undefined result.